theseus/embodied/robot/forward_kinematics.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+
+from typing import List, Optional
+
+from theseus.labs.lie_functional import se3
+from .robot import Robot
+from .link import Link
+from .joint import Joint
+
+
+# TODO: Add support for joints with DOF>1
+
+
+def ForwardKinematicsFactory(robot: Robot, link_names: Optional[List[str]] = None):
+    links: List[Link] = []
+
+    if link_names is None:
+        links = robot.links
```

Comment:
Maybe do `links = robot.links if link_names is None else [robot.link_map[name] for name in link_names]`

Also, I think this is better as a method of `Robot` class. Something like 
```python
def get_links(self, names: Optional[List[str]]) -> List[Link]:
```

Reply:
refactored